pkg/segment/utils/number_test.go

```diff
@@ -0,0 +1,402 @@
+package utils
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Test_isInvalid tests if the number is invalid. It should return invalidType.
+func Test_isInvalid(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+	assert.True(t, n.ntype() == invalidType)
+}
+
+// Test_invalidTypeReturnsErrorForInt64AndFloat64 tests converting invalidType to Int64 and Float64.
+func Test_invalidTypeReturnsErrorForInt64AndFloat64(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+	if val, err := n.Int64(); err == nil || val != 0 {
+		assert.Fail(t, "Expected 0, got %v with error %v", val, err)
+	}
```

Comment:
nit: the way you are doing the `err` and `nil` verification is fine. But you can refactor that code like this:

```
assert.Nil(t, err, "some Message")  // or assert.NoError(t, err, "some message")
assert.Equal(t, 0, val)
```

Same comment elsewhere.

Reply:
changed it to get the variables and then assert.Equal